src/Symfony/Component/DependencyInjection/Compiler/ServiceLocatorTagPass.php

```diff
@@ -47,14 +48,19 @@ protected function processValue($value, bool $isRoot = false)
             $value->setClass(ServiceLocator::class);
         }
 
-        $arguments = $value->getArguments();
-        if (!isset($arguments[0]) || !\is_array($arguments[0])) {
+        $services = $value->getArguments()[0] ?? null;
```

Comment:
Using `getArgument(0)` could throw `OutOfBoundsException` which I consider as BC break. Let me know if it's not.

Reply:
The current code is fine, I guess.